### PR TITLE
Add example tests

### PR DESCRIPTION
**Pull request by vpapanasta**
Hi folks,

I am adding some smoke-level tests for most of our examples. For the examples, I did not add a test, we may need to do some small edits. So I decided to wait.

The set-up method creates an .env file which deletes in a teardown. The .env file will contain the secret key. We also have a teardown to make sure the key environment variable is unset and another one to close any opened figures.

### CODE REVIEW

File: tests/texampleTests.m [hunk 1]
```diff
@@ -0,0 +1,58 @@
+classdef texampleTests < matlab.unittest.TestCase
+% Smoke level tests for the example .mlx files
+
+%   Copyright 2024 The MathWorks, Inc.
+
+    methods (TestClassSetup)
+        function setUpAndTearDowns(testCase)
+            openAIEnvVar = "OPENAI_KEY";
+            key = getenv(openAIEnvVar);
+            writelines("OPENAI_API_KEY="+key,".env");
```
Comment (ccreutzi) [inline tests/texampleTests.m:10]: That seems like a bad idea. I'd rather we used [secrets](https://www.mathworks.com/help/matlab/ref/setsecret.html) or did the `getenv` in the mlx files.

File: tests/texampleTests.m [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+            testCase.addTeardown(@() delete(".env"));
+            testCase.addTeardown(@() unsetenv("OPENAI_API_KEY"));
```
Comment (ccreutzi) [inline tests/texampleTests.m:13]: Running the test should not, I think, modify the environment. If we call `unsetenv` here, no second test file can run in the same MATLAB process and use the env. That sort of (negative) drool can cause hard to debug test dependencies.

File: tests/texampleTests.m [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        end
+    end
+    
+end
+
+function iCloseAll()
+% Close all opened figures
+allFig = findall(0, 'type', 'figure');
+close(allFig)
+end
```
Comment (ccreutzi) [inline tests/texampleTests.m:58]: Or we could just use `close("all")`?

**Review by ccreutzi — CHANGES_REQUESTED** [review]
Haven't started looking into you mlx changes yet, GitHub does not help there.

File: tests/texampleTests.m (hunk 3 above)
Comment (vpapanasta) [reply tests/texampleTests.m:64]: This is the first thing I tried but it does not close `uifigure` which we use in some examples.

File: tests/texampleTests.m (hunk 2 above)
Comment (vpapanasta) [reply tests/texampleTests.m]: I changed the setup method and used a test fixture after a discussion with Christopher.

**Review by ccreutzi — APPROVED** [review]
Checked mlx differences in MATLAB.

**Review by debymf — APPROVED** [review]

**Pull request merged**